src/StreamJsonRpc/Reflection/MessageFormatterProgressTracker.cs

```diff
@@ -0,0 +1,239 @@
+
+        /// <summary>
+        /// Gets or sets the the next id value to assign as token for the progress objects.
+        /// </summary>
+        private long NextProgressId { get; set; }
+
+        /// <summary>
+        /// Converts given <see cref="Type"/> to its <see cref="IProgress{T}"/> type.
+        /// </summary>
+        /// <param name="objectType">The type which may implement <see cref="IProgress{T}"/>.</param>
+        /// <returns>The <see cref="IProgress{T}"/> from given <see cref="Type"/> object, or <c>null</c>  if no such interface was found in the given <paramref name="objectType" />.</returns>
+        public static Type FindIProgressOfT(Type objectType)
+        {
+            Requires.NotNull(objectType, nameof(objectType));
+
+            if (objectType.IsConstructedGenericType && objectType.GetGenericTypeDefinition().Equals(typeof(IProgress<>)))
+            {
+                return objectType;
+            }
+
+            Type iProgressOfTType = null;
+            lock (IProgressOfTTypeMap)
+            {
+                if (!IProgressOfTTypeMap.TryGetValue(objectType, out iProgressOfTType))
+                {
+                    iProgressOfTType = objectType.GetTypeInfo().GetInterfaces().FirstOrDefault(i => i.IsConstructedGenericType && i.GetGenericTypeDefinition() == typeof(IProgress<>));
+                    IProgressOfTTypeMap.Add(objectType, iProgressOfTType);
+                }
+            }
+
+            return iProgressOfTType;
+        }
+
+        /// <summary>
+        /// Saves a <see cref="ProgressParamInformation"/> instance obtained from the given <see cref="object"/> into <see cref="progressMap"/>.
+        /// </summary>
+        /// <param name="value">The object which should implement <see cref="IProgress{T}"/> to create the <see cref="ProgressParamInformation"/> instance.</param>
+        /// <returns>The assigned <see cref="long"/> progres ID.</returns>
+        public long AddProgressObjectToMap(object value)
+        {
+            Requires.NotNull(value, nameof(value));
+
+            if (this.RequestIdBeingSerialized == null)
+            {
+                throw new NotSupportedException(Resources.ProgressObjectInResponseOrNotificationError);
+            }
+
+            lock (this.progressLock)
+            {
+                long progressId = this.NextProgressId++;
+                this.requestProgressMap.Add(this.RequestIdBeingSerialized.Value, progressId);
+
+                this.progressMap.Add(progressId, new ProgressParamInformation(value));
+
+                return progressId;
+            }
+        }
+
+        /// <summary>
+        /// Clears the objects in <see cref="requestProgressMap"/> and <see cref="progressMap"/> associated to the given request ID.
+        /// </summary>
+        /// <param name="requestId">The id of the request whose associated object need to be cleared.</param>
+        public void ClearProgressMaps(long requestId)
```

Comment:
>ClearProgressMaps [](start = 20, length = 17)

This method name describes more of the internals of what it does than what the outside caller should have to think about. Can you name this based on the use case for the caller? For example: `OnResponseReceived`
Then modify the xml doc comments similar to my advice in the above comment.